### PR TITLE
CSV output

### PR DESCRIPTION
**Pull request by biddisco**
1) Add csv output (and launch info) to supported miniapps

    When enabled, extra csv values are displayed using a format that
    includes column titles as well as values.
    
    This makes it easier to use a general purpose post-processing script
    to import data into a pandas table and filter/plot different parameters
    against each other. Different miniapps will have different columns
    but a single script can read any of them and treat them accordingly.
    
    Add a "--info=" command line param so that settings can be passed
    from a launch script into the benchmark output

    An example might look as follows:
    CSVData-2, run, 19, time, 0.145857964, GFlops, 157.04656514561887, type,
      Double, UpLo, Lower, matrixsize, 4096, blocksize, 256, comm_rows, 1,
      comm_cols, 1, threads, 8, backend, GPU,
      machine, daint, branchname, new_polling, ranks_per_node, 4

2) Add a max-seconds param to supported miniapps

    This enables benchmarks to set number of runs to a large number
    but stop after 1 minute, or longer.
    
    This is useful when running many benchmarks with different params that
    might take varying (unpredictable) amounts of time.

### CODE REVIEW

File: miniapp/miniapp_cholesky.cpp [hunk 1]
```diff
@@ -157,15 +173,43 @@ struct choleskyMiniapp {
...
                   << " " << elapsed_time << "s"
                   << " " << gigaflops << "GFlop/s"
                   << " " << dlaf::internal::FormatShort{opts.type}
                   << dlaf::internal::FormatShort{opts.uplo} << " " << matrix_host.size() << " "
                   << matrix_host.blockSize() << " " << comm_grid.size() << " "
                   << pika::get_os_thread_count() << " " << backend << std::endl;
-
+#ifdef DLAF_MINIAPP_CSV_OUTPUT
+        // clang-format off
+        // CSV formatted output with column names that can be read by pandas to simplify post-processing
```
[the remark this answers is not in the document]
Comment (biddisco) [reply miniapp/miniapp_cholesky.cpp:170]: The advantage of using this form of CSV output is that there is no need to parse directory names to get certain options into post-processing scripts. By writing out column names an allowing pp-info to pass in other info that is not available in the original output, you can read a single line of CSV from many minapp directories and automatically generate dataframes with all values needed. I used to parse dir names to get info, but abandoned that in favour of using enhanced CSV. 

I can upgrade your launch and post processing scripts to use this CSV, and deprecate the old if necessary (as another PR)

[the remark this answers is not in the document]
Comment (rasolca) [reply miniapp/miniapp_cholesky.cpp]: For me you can add the optional csv output, but I would rather keep our scripts as they are.
Let's see what the others think about it.

[the remark this answers is not in the document]
Comment (biddisco) [reply miniapp/miniapp_cholesky.cpp]: > For me you can add the optional csv output, but I would rather keep our scripts as they are.
> Let's see what the others think about it.

If I can get the now simplified PR merged, which only add the `--csv` option and `--pp-info` to allow me to pass extra params through, then I'd be very happy. What do I need to do to get this merged?

[the remark this answers is not in the document]
Comment (rasolca) [reply miniapp/miniapp_cholesky.cpp]: A review from someone else.

**Comment by rasolca** [comment]
cscs-ci run

**Review by rasolca — APPROVED** [review]

**Pull request merged**